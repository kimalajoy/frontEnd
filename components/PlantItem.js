import React from 'react';
import { View, Text, Image, TouchableHighlight} from 'react-native';

import styles from '../styles/styles';

export default function PlantItem ({ title, image, navigation }) {
  return (
    <TouchableHighlight
      onPress={() => {
        navigation.navigate('PlantInfoPage');
        navigation.dangerouslyGetParent().setOptions({ tabBarVisible: false });
      }} >
      <View>
        <Image style={styles.meetAPlant} source={{uri: image}}/>
        <Text>{ title }</Text>
      </View>
    </TouchableHighlight>
  );
}
