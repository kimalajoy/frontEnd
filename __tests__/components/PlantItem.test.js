import React from 'react';
import { render, cleanup, fireEvent } from 'react-native-testing-library';

import PlantItem from '../../components/PlantItem';

describe("Plant Item", () => {
  let plantItem;

  beforeEach(() => {

    plantItem = render(<PlantItem
        title={'Test Title'}
      />)
  });

  afterEach(cleanup);

  test('it renders title', () => {
    let title = plantItem.getByText('Test Title')

    expect(title).toBeTruthy()
  })

  test('it renders image', () => {
    let image = plantItem.getByTestId('PlantItem.Image')

    expect(image).toBeTruthy()
  })

  test('it renders / snapshot test', () => {
    expect(plantItem).toMatchSnapshot()
  })
})
