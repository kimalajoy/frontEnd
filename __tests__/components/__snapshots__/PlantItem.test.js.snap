// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plant Item it renders / snapshot test 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 254, 254, 0.8)",
          "marginTop": 30,
          "padding": 20,
          "width": "80%",
        },
        Object {
          "borderRadius": 5,
        },
        Object {
          "elevation": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 24,
          "paddingBottom": 20,
        }
      }
    >
      Test Title
    </Text>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "height": 200,
            "width": 250,
          },
          Object {
            "borderRadius": 5,
          },
        ]
      }
      testID="PlantItem.Image"
    />
  </View>
</View>
`;
